metals/src/main/scala/scala/meta/internal/builds/SbtBuildTool.scala

```diff
@@ -51,31 +51,56 @@ case class SbtBuildTool(version: String) extends BuildTool {
           sbtArgs
         ).flatten
     }
-    writeSbtMetalsPlugin(config)
+    removeOldPlugin()
+    writeSbtMetalsPlugin(workspace, config)
     allArgs
   }
+
   override def digest(
       workspace: AbsolutePath
   ): Option[String] = SbtDigest.current(workspace)
   override val minimumVersion: String = "0.13.17"
 
-  private def writeSbtMetalsPlugin(config: MetalsServerConfig): Unit = {
+  private def removeOldPlugin(): Unit = {
+    def pluginsDirectory(version: String): AbsolutePath = {
+      AbsolutePath(System.getProperty("user.home"))
+        .resolve(".sbt")
+        .resolve(version)
+        .resolve("plugins")
+    }
     val plugins =
-      if (version.startsWith("0.13")) SbtBuildTool.pluginsDirectory("0.13")
-      else SbtBuildTool.pluginsDirectory("1.0")
-    plugins.createDirectories()
+      if (version.startsWith("0.13")) pluginsDirectory("0.13")
+      else pluginsDirectory("1.0")
+
+    val metalsFile = plugins.resolve("metals.sbt")
+    if (metalsFile.exists) metalsFile.delete()
+  }
+
+  private def writeSbtMetalsPlugin(
+      workspace: AbsolutePath,
+      config: MetalsServerConfig
+  ): Unit = {
     val bytes = SbtBuildTool
-      .globalMetalsSbt(config.bloopSbtVersion)
+      .sbtPlugin(config.bloopSbtVersion)
       .getBytes(StandardCharsets.UTF_8)
-    val destination = plugins.resolve("metals.sbt")
-    if (destination.isFile && destination.readAllBytes.sameElements(bytes)) {
+    val projectDir = workspace.resolve("project")
+    projectDir.toFile.mkdir()
+    val metalsPluginfile = projectDir.resolve("metals.sbt")
+    if (metalsPluginfile.isFile && metalsPluginfile.readAllBytes.sameElements(
+        bytes
+      )) {
       // Do nothing if the file is unchanged. If we write to the file unconditionally
       // we risk triggering sbt re-compilation of global plugins that slows down
       // build import greatly. If somebody validates it doesn't affect load times
       // then feel free to remove this guard.
       ()
     } else {
-      Files.write(destination.toNIO, bytes)
+      Files.write(metalsPluginfile.toNIO, bytes)
+    }
+    val gitignore = workspace.resolve(".gitignore")
```

Comment:
maybe extract the gitignore-related stuff into a method?

Reply:
added it to a method and also added some tests for it